Automattic/jetpack
# Added instanceof WP_Post check

Fixes #16835
>In some scenarios, the following notice can be triggered:
>>PHP Notice: Trying to get property 'post_author' of non-object in wp-content/plugins/jetpack/class.jetpack-post-images.php on line 528

#### Changes proposed in this Pull Request:

Added instanceof WP_Post check in `from_gravatar` function to avoid accessing post_author of a non-object. If that is the case, the function will return an empty array similar to what other functions in the file do (`from_blocks`, `from_images`, `from_blavatar`...)


#### Other information:

- [ X ] Have you written new tests for your changes, if applicable?
- [ X ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
No

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:





Haven't been able to reproduce the issue with the normal flow of the Application so needed to do some **tweaking**. The easiest is through Related Posts. Top posts widget also seemed to go through the flow, but I have been unable to test it that way.
* Enable related posts in Jetpack 
![image](https://user-images.githubusercontent.com/13239096/186658088-becb8b30-95ef-4378-a642-0f54eac32f61.png)
* Create two posts without images attached or galleries and same tags so that they will appear as related. (It may take a while to process the related posts) 
* Go to the first post created and check that related posts are available. One of them should be the second post, and it should not show any images, it should show the text instead.
* In the `get_images` method, set to **true** `fallback_to_avatars` in the `defaults` array (line 664). This way at the end of the method, the from_gravatar method will be called if no images are associated with the related post.
* Set post to null by switching `$post = get_post( $post_id );` with `$post = null;` at the beginning of `from_gravatar` method.
With these instructions and the code added in the commit, you should not see the Notice or Warning (depending on PHP Version running). If code from the commit is commented out, then the Notice or Warning will be available.
